cmd/kubeadm/app/apis/kubeadm/validation/validation_test.go

```diff
@@ -102,15 +102,12 @@ func TestValidateNodeRegistrationOptions(t *testing.T) {
 		criSocket      string
 		expectedErrors bool
 	}{
-		{"", "/some/path", true},                                                             // node name can't be empty
-		{"INVALID-NODENAME", "/some/path", true},                                             // Upper cases is invalid
-		{"invalid-nodename-", "/some/path", true},                                            // Can't have trailing dashes
-		{"invalid-node?name", "/some/path", true},                                            // Unsupported characters
-		{"valid-nodename", "/some/path", false},                                              // supported
-		{"valid-nodename-with-numbers01234", "/some/path/with/numbers/01234/", false},        // supported, with numbers as well
-		{"valid-nodename", kubeadmapiv1beta2.DefaultUrlScheme + "://" + "/some/path", false}, // supported, with socket url
-		{"valid-nodename", "bla:///some/path", true},                                         // unsupported url scheme
-		{"valid-nodename", ":::", true},                                                      // unparseable url
+		{"", "/some/path", true},                  // node name can't be empty
+		{"INVALID-NODENAME", "/some/path", true},  // Upper cases is invalid
+		{"invalid-nodename-", "/some/path", true}, // Can't have trailing dashes
+		{"invalid-node?name", "/some/path", true}, // Unsupported characters
+		{"valid-nodename", "/some/path", false},   // supported
+		{"valid-nodename", "", true},              // invalid CRISocket (path is not absolute)
```

Comment:
the test case `// invalid CRISocket (path is not absolute)` should be moved to TestValidateSocketPath

if TestValidateNodeRegistrationOptions now unit tests only the node name, then the struct field `criSocket` should be removed and `/some/path` should always be passed to `ValidateNodeRegistrationOptions`

Reply:
So my thinking there was - Do we need to ensure that `ValidateNodeRegistrationOptions` calls `ValidateSocketPath` for the CRISocket?
The test case is also part of TestValidateSocketPath.

Reply:
> Do we need to ensure that ValidateNodeRegistrationOptions calls ValidateSocketPath for the CRISocket?

this is also a generic question whether we should unit test A, B, C if A calls B and C.
it is enough to unit test A with all the cases?
the answer is yes, if testing A doesn't result in too many test cases.

so it is better to not add a separate function TestValidateSocketPath.
given the number of test cases is not that big for ValidateNodeRegistrationOptions and everything can be there.

so you can re-open if you'd like:
https://github.com/kubernetes/kubernetes/pull/91393
or just rework this PR to just add the missing cases.

also golang supports unit test coverage (if you haven't checked that already):
https://blog.golang.org/cover

Reply:
Wouldn't it be easier to have a comprehensive list of test cases for C, then in B just make sure C is called? That would make A much easier to maintain in the future. Otherwise, it is only a matter of time before we feel that A has too many test cases. The `TestValidateInitConfiguration`, `TestValidateClusterConfiguration` and `TestValidateJoinConfiguration` are probably already close to that state because there is a lot of nested calls in there.

Reply:
we do have plans to extend NodeRegistrationOptions in the future, so "number of test cases" is a valid concern.

but note that this test case does not ensure that ValidateNodeRegistrationOptions calls ValidateSocketPath, it ensures that "something" returns an error on empty socket.

i'd just remove this test case to not duplicate it and add a comment that TestValidateNodeRegistrationOptions only validates node names, while socket names are tested in TestValidateSocketPath

Reply:
>i'd just remove this test case to not duplicate it and add a comment that TestValidateNodeRegistrationOptions only validates node names, while socket names are tested in TestValidateSocketPath

SGTM

>but note that this test case does not ensure that ValidateNodeRegistrationOptions calls ValidateSocketPath, it ensures that "something" returns an error on empty socket.

Fair enough. But then wouldn't we want _something_ to make sure `ValidateNodeRegistrationOptions` returns an error on empty socket?

Reply:
Pushed up another revision based on your suggestions.

Reply:
> Fair enough. But then wouldn't we want something to make sure ValidateNodeRegistrationOptions returns an error on empty socket?

one way is to use error string matching, but let's not do that here.
the change is fine as is.

added on comment bellow.